src/managed/Microsoft.NET.HostModel/ComHost/ComHost.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.NET.HostModel.ComHost
+{
+    class ComHost
+    {
+        // These need to match RESOURCEID_CLISDMAP and RESOURCETYPE_CLSIDMAP defined in comhost.h.
+        private const int ClsidmapResourceId = 64;
+        private const int ClsidmapResourceType = 1024;
+
+        /// <summary>
+        /// Create an ComHost with an embedded CLSIDMap file to map CLSIDs to .NET Classes.
+        /// </summary>
+        /// <param name="comHostSourceFilePath">The path of Apphost template, which has the place holder</param>
+        /// <param name="comHostDestinationFilePath">The destination path for desired location to place, including the file name</param>
+        /// <param name="clsidmapFilePath">The path to the *.clsidmap file.</param>
+        public static void Create(
+            string comHostSourceFilePath,
+            string comHostDestinationFilePath,
+            string clsidmapFilePath)
+        {
+            var destinationDirectory = new FileInfo(comHostDestinationFilePath).Directory.FullName;
+            if (!Directory.Exists(destinationDirectory))
+            {
+                Directory.CreateDirectory(destinationDirectory);
+            }
+
+            // Copy apphost to destination path so it inherits the same attributes/permissions.
+            File.Copy(comHostSourceFilePath, comHostDestinationFilePath, overwrite: true);
+
+            if (ResourceUpdater.IsSupportedOS())
```

Comment:
Flip this predicate and simply throw an exception. The current contents should be moved to the end of the function no need for an `else` clause.